### PR TITLE
MSA Pallet Docs

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to improve the documentation of the Pallets and make that documentation be able to be used on `docs.frequency.xyz`.

Part of https://github.com/frequency-chain/docs/issues/59

# Discussion
- This is the first one, so it includes the needed documentation and templates as well.
- Please suggest additional content that would be useful as well
- Had a hard time getting it to look nice in both markdown and the rendered html.
- Disabling soft wrap will help it look better
- The idea was to keep rust specific things out of the readme and in the `lib.rs`

# Testing
- `make docs`
- `open target/doc/pallet_msa/index.html`

## Screenshots

image

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -197,7 +197,7 @@ benchmarks-capacity:
 
 .PHONY: docs
 docs:
-	RUSTDOCFLAGS="--enable-index-page -Zunstable-options" cargo doc --no-deps --features frequency
+	RUSTC_BOOTSTRAP=1 RUSTDOCFLAGS="--enable-index-page -Zunstable-options" cargo doc --no-deps --features frequency
```
Comment (wilwade) [inline Makefile:200]: Needed for some versions of rust, but fine on all as far as I can tell

File: pallets/msa/README.md
Comment (wilwade) [inline pallets/msa/README.md:1]: Try reviewing in "Rich diff" 
image

File: pallets/msa/src/lib.rs [hunk 2]
```diff
@@ -1,35 +1,13 @@
-//! # MSA Pallet
-//! The MSA pallet provides functionality for handling Message Source Accounts.
+//! Message Source Account Management
```
Comment (wilwade) [inline pallets/msa/src/lib.rs:1]: This is here as it is what is picked up by rust docs to show as the short description in the docs search

File: tools/scripts/show-runtime-version.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+# Quick script to output the runtime spec_version at a particular commit.
```
Comment (wilwade) [inline tools/scripts/show-runtime-version.sh:3]: Helpful for getting the spec_version in the past. I can drop this file if people think it isn't worth it.

**Review by saraswatpuneet — APPROVED** [review]
Neat

File: pallets/msa/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+All users on Frequency must have an MSA in order to:
+
+1. Acquire a User Handle
+2. Delegate tasks to Providers (defining specific tasks for specific providers)
+3. Become a Provider so they may do Provider level tasks on their own behalf
+4. Have Message or Stateful Storage data
+
+### MSA Id and Keys
+
+Once a user creates an MSA, they are assigned an MSA Id–a unique number permanently assigned to the number at the time of creation as well as one or more control keys attached for control.
```
Comment (JoeCap08055) [inline pallets/msa/README.md:17]:
```suggestion
Once a user creates an MSA, they are assigned an MSA Id–a unique number permanently assigned to the user at the time of creation as well as one or more control keys attached for control.
```

Not sure if "to the user" is correct, either... maybe just remove the phrase?

File: pallets/msa/src/lib.rs [hunk 5]
```diff
@@ -39,12 +17,10 @@
...
 //!
-//! ### Assumptions
-//!
-//! * Total MSA keys should be less than the constant `Config::MSA::MaxPublicKeysPerMsa`.
-//! * Maximum schemas, for which provider has publishing rights, be less than `Config::MSA::MaxSchemaGrantsPerDelegation`
+//! ## Assumptions
 //!
-// Substrate macros are tripping the clippy::expect_used lint.
+//! - Total MSA keys should be less than the constant `Config::MSA::MaxPublicKeysPerMsa`.
+//! - Maximum schemas, for which provider has publishing rights, be less than `Config::MSA::MaxSchemaGrantsPerDelegation`
```
Comment (JoeCap08055) [inline pallets/msa/src/lib.rs:23]:
```suggestion
//! - Maximum schemas, for which any single provider has publishing rights on behalf of a single user, must be less than `Config::MSA::MaxSchemaGrantsPerDelegation`
```

File: tools/scripts/show-runtime-version.sh (hunk 3 above)
Comment (JoeCap08055) [reply tools/scripts/show-runtime-version.sh:3]: 💯

**Review by JoeCap08055 — APPROVED** [review]

**Review by JoeCap08055 — APPROVED** [review]
A few suggestions, but non-blocking

File: pallets/msa/README.md (hunk 4 above)
Comment (wilwade) [reply pallets/msa/README.md]: Updated to:

```
Once a user creates an MSA, they are assigned an MSA Id, a unique number the time of creation with one or more keys attached for control.
```

**Comment by jeanettedepatie** [comment]
In my view, the Extrinsics table is cut off on the bottom.

**Review by jeanettedepatie — COMMENTED** [review]
Also, the links going to the rust docs don't seem to work for me.  Not sure if I'm improperly configured.  Otherwise looks good.

File: pallets/msa/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+| --------------------------------------------------------------------------------------------- | ---------------------------------------- | ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------- |
+| `add_public_key_to_msa`<br />Add MSA control key                                              | Delegator or Provider with Signature     | Free               | [`PublicKeyAdded`](https://frequency-chain.github.io/frequency/pallet_msa/pallet/enum.Event.html#variant.PublicKeyAdded)                                                                                                                         | 1             |
```
Comment (mattheworris) [inline pallets/msa/README.md:34]: nit: Depending on the audience, is `Free` ambiguous? e.g. Free for end users, who are using Capacity. Not sure if this is a good spot to make the distinction.

**Review by mattheworris — APPROVED** [review]
- One non-blocking comment
- Nice improvements

🚢 it!

File: pallets/msa/README.md (hunk 6 above)
Comment (wilwade) [reply pallets/msa/README.md:34]: Actually this line is wrong! This one is supposed to be `Capacity` which is separate from `Free`. The `Caller` column was supposed to help disambiguate Capacity and free.

**Review by shannonwells — APPROVED** [review]
looks great, thanks!
* I reviewed the code
* I read the markdown
* I generated the docs and looked at some of the pages.

**Pull request merged**